grails-app/services/org/pih/warehouse/receiving/ReceiptService.groovy

```diff
@@ -248,6 +249,19 @@ class ReceiptService {
             }
         }
 
+        // Send notification email on completed receiving
+        if (completed) {
+            def g = grailsApplication.mainContext.getBean('org.codehaus.groovy.grails.plugins.web.taglib.ApplicationTagLib')
+            def recipientItems = partialReceipt.partialReceiptItems.groupBy {it.recipient }
+            recipientItems.each {recipient, items ->
+                def subject = g.message(code: "email.yourItemReceived.message", args: [shipment.shipmentNumber])
+                def body = "${g.render(template: "/email/shipmentItemReceived", model: [shipmentInstance: shipment, receiptItems: items])}"
+                if (recipient) {
```

Comment:
Move this up a few lines so we don't need to render the template unless there's a recipient.